### PR TITLE
files:scan add number of items per second on printed result table

### LINKED ISSUE #32085: [Feature Request] Improve console output for files scan command
10.0.9

Running ``sudo -uwww-data ./occ files:scan --path="user_id/files/mount_point"``

Scanning files for 1 users
Starting scan for user 1 out of 1 (user_id)

```
+---------+-------+----------+
| Folders | Files | Elapsed  |
+---------+-------+----------+
| 519     | 19794 | 00:02:34 |
+---------+-------+----------+
```

**Feature Request**
Adding a Items per second column (folders + files / used seconds)
```
+---------+-------+----------+------------------+
| Folders | Files | Elapsed  | Items per second |
+---------+-------+----------+------------------+
| 519     | 19794 | 00:02:34 |        260       |
+---------+-------+----------+------------------+
```

**Benefit:**
Performance overview

### PR DESCRIPTION
**Pull request by mmattel**
## Description

This PR adds to the table printout of files:scan the items per second processed

## Related Issue

- Fixes #32085 ([Feature Request] Improve console output for files scan command)

## Motivation and Context

Prints a calculated performance.
Note: on low number of seconds the result may differ a bit because fractions of a second are calculated but only seconds are shown in the table

## How Has This Been Tested?

```
sudo -uwww-data ./occ files:scan --path="user_1/files/mount_folder_1"

Scanning files for 1 users
Starting scan for user 1 out of 1 (user_1)

+---------+-------+--------------+------------------+
| Folders | Files | Elapsed time | Items per second |
+---------+-------+--------------+------------------+
| 60      | 863   | 00:00:06     | 149              |
+---------+-------+--------------+------------------+

sudo -uwww-data ./occ files:scan --path="user_1/files/mount_folder_2"

Scanning files for 1 users
Starting scan for user 1 out of 1 (user_1)

+---------+-------+--------------+------------------+
| Folders | Files | Elapsed time | Items per second |
+---------+-------+--------------+------------------+
| 1027    | 14903 | 00:02:57     | 169              |
+---------+-------+--------------+------------------+
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by mmattel** [comment]
@phil-davis can your restart drone, many thanks (Error: Client.Timeout exceeded while awaiting headers)

File: apps/files/lib/Command/Scan.php [hunk 1]
```diff
@@ -444,11 +444,13 @@ protected function presentStats(OutputInterface $output) {
 	 */
 	protected function showSummary($headers, $rows, OutputInterface $output) {
 		$niceDate = $this->formatExecTime();
+		$items_per_second = $this->getItemsPerSecond();
```
Comment (PVince81) [inline apps/files/lib/Command/Scan.php]: please use camel case (I hope php-cs-fixer spots this!)

File: apps/files/lib/Command/Scan.php [hunk 2]
```diff
@@ -458,6 +460,22 @@ protected function showSummary($headers, $rows, OutputInterface $output) {
...
+	 */
+	protected function getItemsPerSecond() {
+		$items = $this->foldersCounter + $this->filesCounter;
+		if ($this->execTime === 0) {
+			// catch div by 0
+			$items_per_second = 0;
+		} else {
+			$items_per_second = $items / $this->execTime;
+		}
+		return \sprintf("%.0f", $items_per_second);
```
Comment (PVince81) [inline apps/files/lib/Command/Scan.php]: I suggest returning the float value here and format it later when adding to the table, also add a unit to the formatted text, for example `"%.0f items/seconds"`

**Review by PVince81 — COMMENTED** [review]
Nice idea, see comments.

File: apps/files/lib/Command/Scan.php (hunk 2 above)
Comment (mmattel) [reply apps/files/lib/Command/Scan.php]: I followed the same way to return / print the time used. It returns the final string to be printed.
Adding the unit does not make sense, as the table header already says what it is: ``Items per second``.
You can see it in the example printouts.

File: apps/files/lib/Command/Scan.php (hunk 1 above)
Comment (mmattel) [reply apps/files/lib/Command/Scan.php]: thx, just fixing it

File: apps/files/lib/Command/Scan.php (hunk 2 above)
Comment (PVince81) [reply apps/files/lib/Command/Scan.php]: as yes, I missed the headers. nice

**Comment by PVince81** [comment]
From what I see adding a unit test assertion would be tricky due to the time element and not that useful.

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**